pkg/asset/internal/templates.go

```diff
@@ -913,21 +913,71 @@ spec:
     - --name=boot-etcd
     - --listen-client-urls=http://0.0.0.0:12379
     - --listen-peer-urls=http://0.0.0.0:12380
-    - --advertise-client-urls=http://$(MY_POD_IP):12379
-    - --initial-advertise-peer-urls=http://$(MY_POD_IP):12380
-    - --initial-cluster=boot-etcd=http://$(MY_POD_IP):12380
+    - --advertise-client-urls=http://{{ .BootEtcdServiceIP }}:12379
```

Comment:
IIUC the underlying issue is that if these flags need to match how the etcd-operator reaches out to the seed-etcd, otherwise it will fail.

If we want to be able to render all the manifests we need for bootstrap (prior to ever running `bootkube start`), then we won't know what this IP will end up as - and can't pre-generate / populate the etcd TPR here: https://github.com/kubernetes-incubator/bootkube/blob/master/pkg/util/etcdutil/migrate.go#L143

The service ip would allow us to use a consistent address between the etcd-operator and the seed so that we are able to pre-render all of the needed assets.

@hongchaodeng open to other ideas, but (assuming my understanding is correct) leaving this using the downward api means that the address populated in the etcd TPR will be incorrect.

The only other option I could see would be to enforce that the etcd-operator run on the same node as the bootstrap (or run a temp bootstrap etcd-operator as static pod) and then just use localhost?